activerecord/lib/active_record.rb

```diff
@@ -339,6 +339,14 @@ def self.global_executor_concurrency # :nodoc:
   singleton_class.attr_accessor :query_transformers
   self.query_transformers = []
 
+  def self.load_tree(enable:)
+    ActiveSupport::IsolatedExecutionState[:active_record_load_tree_on] = enable
```

Comment:
Why is this an isolate state? Is the idea that each thread will have this config enabled independently? This config sounds like a regular global which you enable in your entire application.

Reply:
Other settings I have added needed this, I was not sure what was appropriate here. I can change it, is there a certain pattern I should use?